### PR TITLE
process all files generated by file generated jar

### PR DESCRIPTION
**Pull request by MauricioUyaguari**

### CODE REVIEW

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 1]
```diff
@@ -35,15 +35,15 @@
...
 import static org.finos.legend.depot.tracing.resources.ResourceLoggingAndTracing.GET_REVISION_FILE_GENERATION_ENTITIES;
 import static org.finos.legend.depot.tracing.resources.ResourceLoggingAndTracing.GET_VERSION_FILE_GENERATION;
 import static org.finos.legend.depot.tracing.resources.ResourceLoggingAndTracing.GET_VERSION_FILE_GENERATION_BY_FILEPATH;
-import static org.finos.legend.depot.tracing.resources.ResourceLoggingAndTracing.GET_VERSION_FILE_GENERATION_BY_PATH;
+import static org.finos.legend.depot.tracing.resources.ResourceLoggingAndTracing.GET_VERSION_FILE_GENERATION_BY_ELEMENT_PATH;
 import static org.finos.legend.depot.tracing.resources.ResourceLoggingAndTracing.GET_VERSION_FILE_GENERATION_ENTITIES;
 
 @Path("")
-@Api("File Generations")
+@Api("Generations")
```
[the remark this answers is not in the document]
Comment (MauricioUyaguari) [reply legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:46]: as discussed will move to a github issue

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 2]
```diff
@@ -57,20 +54,20 @@ public FileGenerationsResource(FileGenerationsService generationsService)
 
     @GET
     @Path("/projects/{groupId}/{artifactId}/latest/generations")
-    @ApiOperation(GET_REVISION_FILE_GENERATION_ENTITIES)
+    @ApiOperation(GET_VERSION_FILE_GENERATION_ENTITIES)
```
Comment (elopezcastro) [inline legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:57]: this should be revision

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 3]
```diff
@@ -57,20 +54,20 @@ public FileGenerationsResource(FileGenerationsService generationsService)
...
     @Produces(MediaType.APPLICATION_JSON)
     public List<Entity> getLatestGenerations(@PathParam("groupId") String groupId,
                                              @PathParam("artifactId") String artifactId)
     {
         QueryMetricsContainer.record(groupId, artifactId, VersionValidator.MASTER_SNAPSHOT);
-        return handle(GET_REVISION_FILE_GENERATION_ENTITIES, () -> this.generationsService.getLatestGenerations(groupId, artifactId));
+        return handle(GET_VERSION_FILE_GENERATION_ENTITIES, () -> this.generationsService.getLatestGenerations(groupId, artifactId));
```
Comment (elopezcastro) [inline legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:63]: this should be revision

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 4]
```diff
@@ -80,37 +77,37 @@ public List<Entity> getLatestGenerationsEntities(@PathParam("groupId") String gr
 
     @GET
     @Path("/generations/{groupId}/{artifactId}/latest")
-    @ApiOperation(GET_REVISION_FILE_GENERATION)
+    @ApiOperation(GET_VERSION_FILE_GENERATION)
```
Comment (elopezcastro) [inline legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:80]: this should be revision

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 5]
```diff
@@ -80,37 +77,37 @@ public List<Entity> getLatestGenerationsEntities(@PathParam("groupId") String gr
...
     @Produces(MediaType.APPLICATION_JSON)
-    public List<FileGeneration> getFileGenerationEntities(@PathParam("groupId") String groupId,
+    public List<FileGeneration> getLatestFileGenerations(@PathParam("groupId") String groupId,
                                                           @PathParam("artifactId") String artifactId)
     {
         QueryMetricsContainer.record(groupId, artifactId, VersionValidator.MASTER_SNAPSHOT);
-        return handle(GET_REVISION_FILE_GENERATION, () -> this.generationsService.getLatestFileGenerations(groupId, artifactId));
+        return handle(GET_VERSION_FILE_GENERATION, () -> this.generationsService.getLatestFileGenerations(groupId, artifactId));
```
Comment (elopezcastro) [inline legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:86]: this should be revision

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 6]
```diff
@@ -80,37 +77,37 @@ public List<Entity> getLatestGenerationsEntities(@PathParam("groupId") String gr
...
     }
 
     @GET
-    @Path("/generations/{groupId}/{artifactId}/latest/{path}")
-    @ApiOperation(GET_REVISION_FILE_GENERATION_BY_PATH)
+    @Path("/generations/{groupId}/{artifactId}/latest/{elementPath}")
+    @ApiOperation(GET_VERSION_FILE_GENERATION_BY_ELEMENT_PATH)
```
Comment (elopezcastro) [inline legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:91]: this should be revision

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 7]
```diff
@@ -80,37 +77,37 @@ public List<Entity> getLatestGenerationsEntities(@PathParam("groupId") String gr
...
     @Produces(MediaType.APPLICATION_JSON)
-    public List<FileGeneration> getLatestFileGenerationsByPath(@PathParam("groupId") String groupId,
+    public List<FileGeneration> getLatestFileGenerationsByElementPath(@PathParam("groupId") String groupId,
                                                                @PathParam("artifactId") String artifactId,
-                                                               @PathParam("path") String path)
+                                                               @PathParam("elementPath") String elementPath)
     {
         QueryMetricsContainer.record(groupId, artifactId, VersionValidator.MASTER_SNAPSHOT);
-        return handle(GET_REVISION_FILE_GENERATION_BY_PATH, () -> this.generationsService.getLatestFileGenerationsByPath(groupId, artifactId, path));
+        return handle(GET_VERSION_FILE_GENERATION_BY_ELEMENT_PATH, () -> this.generationsService.getLatestFileGenerationsByElementPath(groupId, artifactId, elementPath));
```
Comment (elopezcastro) [inline legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:98]: this should be revision

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 8]
```diff
@@ -80,37 +77,37 @@ public List<Entity> getLatestGenerationsEntities(@PathParam("groupId") String gr
...
     }
 
     @GET
-    @Path("/generations/{groupId}/{artifactId}/latest/file/{file}")
-    @ApiOperation(GET_REVISION_FILE_GENERATION_BY_FILEPATH)
+    @Path("/generations/{groupId}/{artifactId}/latest/file/{filePath}")
+    @ApiOperation(GET_VERSION_FILE_GENERATION_BY_FILEPATH)
```
Comment (elopezcastro) [inline legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:103]: change to revision constant

File: legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java [hunk 9]
```diff
@@ -80,37 +77,37 @@ public List<Entity> getLatestGenerationsEntities(@PathParam("groupId") String gr
...
     @Produces(MediaType.APPLICATION_JSON)
-    public Optional<FileGeneration> getLatestFileGenerationsByFile(@PathParam("groupId") String groupId,
+    public Optional<FileGeneration> getLatestFileGenerationsByFilePath(@PathParam("groupId") String groupId,
                                                                    @PathParam("artifactId") String artifactId,
-                                                                   @PathParam("file") String file)
+                                                                   @PathParam("filePath") String filePath)
     {
         QueryMetricsContainer.record(groupId, artifactId, VersionValidator.MASTER_SNAPSHOT);
-        return handle(GET_REVISION_FILE_GENERATION_BY_FILEPATH, () -> this.generationsService.getLatestFileGenerationsByFile(groupId, artifactId, file));
+        return handle(GET_VERSION_FILE_GENERATION_BY_FILEPATH, () -> this.generationsService.getLatestFileGenerationsByFilePath(groupId, artifactId, filePath));
```
Comment (elopezcastro) [inline legend-depot-server/src/main/java/org/finos/legend/depot/server/resources/file/FileGenerationsResource.java:110]: same here

**Review by elopezcastro — APPROVED** [review]

**Pull request merged**